plugins/content_distribution/providers/facebook/lib/FacebookDistributionEngine.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * @package plugins.facebookDistribution
+ * @subpackage lib
+ */
+class FacebookDistributionEngine extends DistributionEngine implements IDistributionEngineSubmit, IDistributionEngineCloseSubmit, IDistributionEngineUpdate
+{
+	protected $appId;
+	protected $appSecret;
+	
+	/* (non-PHPdoc)
+	 * @see DistributionEngine::configure()
+	 */
+	public function configure()
+	{
+		parent::configure();
+		if (isset(KBatchBase::$taskConfig->params->facebook))
+		{
+			if (isset(KBatchBase::$taskConfig->params->facebook->appId))
+				$this->appId = KBatchBase::$taskConfig->params->facebook->appId;
+			if (isset(KBatchBase::$taskConfig->params->facebook->appSecret))
+				$this->appSecret = KBatchBase::$taskConfig->params->facebook->appSecret;
+		}
+	}
+
+	/* (non-PHPdoc)
+	 * @see IDistributionEngineSubmit::submit()
+	 */
+	public function submit(KalturaDistributionSubmitJobData $data)
+	{
+		$this->validate($data);	
+		if($data->entryDistribution->remoteId)
+		{
+			$data->remoteId = $data->entryDistribution->remoteId;
+		}
+		else 
+		{
+			$this->doSubmit($data, $data->distributionProfile);
+		}
+		return true;
```

Comment:
If submit always returns true then there is no need to implement IDistributionEngineCloseSubmit.